backend/src/apiserver/common/config.go

```diff
@@ -146,6 +146,14 @@ func GetArtifactImage() string {
 	return GetStringConfigWithDefault(ArtifactImage, DefaultArtifactImage)
 }
 
+func GetObjectStoreAccessKey() string {
+	return GetStringConfig(ObjectStoreAccessKey)
```

Comment:
By getting the S3 credential from viper, the whole cluster has to share the same S3 credential for all users. Whereas by getting the S3 credential from the secret, because users has full control of their namespaces so they can update the `mlpipeline-minio-artifact` with their own S3 credential without impacting other users.

Reply:
(this and the following review feedback items are related so centralizing the discussion here)

This makes sense, and it would be good to parameterize these somehow.  As you mentioned below, the easiest path is probably to create some new env vars for the s3 Secret name, accesskey Key, and secretkey Key.  It is a little clunky, but does remove the hardcoded secret name.  However, this would mean every namespace would need to use identical secret names across Namespaces (ie namespace1 and namespace2 can each have a `foo-s3-secret` with different contents, but the secret *must* always be named `foo-s3-secret` and use the same `accesskey` and `secretkey` keys.

I think that's a step in the right direction but it would be better if a namespace can individually reference an arbitrary secret name in a vacuum (namespace1 references `foo-secret` but namespace2 references `bar-secret`).  I'm not sure of a clean way to incorporate that though, you wouldn't happen to know of a better vector for retrieving these parameters than apiserver env vars, would you?  If not I can push a changeset that implements the parameterized secret name as outline above.